From: Modestas Vainius <modax@debian.org>
Subject: workaround FTBFS on powerpc/gcc 4.6 in signalsloteditor
Origin: vendor
Forwarded: not-needed
Last-Update: 2011-05-07

This should workaround g++-4.6/powerpc ICE that is related to dwarf2 [1][2].
It is very likely the patch will not be needed long term as the gcc bug might
eventually be fixed.

[1] .moc/release-shared/qteditorfactory.moc:237:1: internal compiler error: in size_of_die, at dwarf2out.c:10749
Please submit a full bug report,
with preprocessed source if appropriate.

[2] https://buildd.debian.org/status/fetch.php?pkg=qt4-x11&arch=powerpc&ver=4%3A4.7.2-4&stamp=1304320067

---
 tools/designer/src/components/signalsloteditor/signalsloteditor.pri |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/tools/designer/src/components/signalsloteditor/signalsloteditor.pri
+++ b/tools/designer/src/components/signalsloteditor/signalsloteditor.pri
@@ -19,3 +19,10 @@ SOURCES += $$PWD/signalslot_utils.cpp \
     $$PWD/signalsloteditorwindow.cpp
 
 FORMS += $$PWD/connectdialog.ui
+
+equals(QT_ARCH, powerpc) {
+    message("powerpc/gcc 4.6 ICE workaround for designer: build with -gstabs rather than -g")
+    QMAKE_CXXFLAGS -= -g
+    QMAKE_CXXFLAGS += -gstabs
+}
+
